.github/workflows/core-ci.yml

```diff
@@ -0,0 +1,33 @@
+name: core-build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "src/**"
```

Comment:
You have to include the workfow itself I think.

Reply:
Ah, right. Good call.

Reply:
Added the change as requested.